tachyon/math/matrix/matrix_utils.h

```diff
@@ -72,6 +72,35 @@ std::vector<PackedField*> PackRowHorizontally(
   });
 }
 
+// Creates a vector of packed fields for a given matrix row. If the length
+// of the row is not a multiple of |PackedField::N|, the last |PackedField|
+// element populates leftover values with |F::Zero()|.
+template <typename Expr, int BlockRows, int BlockCols, bool InnerPanel,
+          typename PackedField =
+              typename PackedFieldTraits<typename Expr::Scalar>::PackedField>
+std::vector<PackedField> PackRowHorizontallyPadded(
+    const Eigen::Block<Expr, BlockRows, BlockCols, InnerPanel>& matrix_row) {
+  size_t cols = static_cast<size_t>(matrix_row.cols());
+  using F = typename FiniteFieldTraits<PackedField>::PrimeField;
+  size_t num_full_packed = cols / PackedField::N;
+  std::vector<PackedField> ret(num_full_packed + 1);
+  OPENMP_PARALLEL_FOR(size_t col = 0; col < num_full_packed; ++col) {
```

Comment:
Thought it could be parallelized since the concerning calculations inside are separate; however, I now see the total times its run is pretty small and not worth parallelizing. I'll change to not parallelized